### PR TITLE
Stats: Polish post detail tables with existing stats table styles

### PR DESCRIPTION
**Pull request by dognose24**
#### Proposed Changes

* Polish and reuse the existing stats table and control styles on the table section of the post detail page.
* Gate the featured changes on `PostMonths` and `PostWeeks` by the flag `stats/enhance-post-detail`.
* Extract the shared stats table styles as a separate file.

#### Testing Instructions

> For a11n, if you have no idea about the testing data on the tables, please refer to `p1671729060450249/1671728562.298969-slack-C0438NHCLSY`.

* Spin up this change with the Calypso Live link.
* Navigate to the Stats `Post Detail` page (`/stats/post/${post-id}/${site-id}`).
* Ensure the post detail page works as previously.
* Append the feature flag `?flags=stats/enhance-post-detail` to the URL.
* Ensure the post detail page table styles are polished in line with the design.

##### Months and years table

| Before | After |
| --- | --- |
| total_before | total_after_3 |

##### Average per day table

| Before | After |
| --- | --- |
| average_before | average_after_3 |

##### Recent weeks table

| Before | After |
| --- | --- |
| week_before | week_after_3 |

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #70671

### CODE REVIEW

**Review by jeffgolenski — APPROVED** [review]
Good find. Looks good my friend

**Review by keoshi — APPROVED** [review]
Looks great to me, thanks for being accommodating with the changes! 🙌

File: client/my-sites/stats/post-detail-table-section/index.tsx [hunk 1]
```diff
@@ -24,33 +25,58 @@ export default function PostDetailTableSection( {
 	const tableOptions = [
 		{ value: 'years', label: translate( 'Months and years' ) },
 		{ value: 'averages', label: translate( 'Average per day' ) },
-		{ value: 'weeks', label: translate( 'Recent Weeks' ) },
+		{ value: 'weeks', label: translate( 'Recent weeks' ) },
```
Comment (jsnmoon) [inline client/my-sites/stats/post-detail-table-section/index.tsx:28]: 😬 I know this is a necessary change, but let's avoid making mistakes like this in the future.

**Review by jsnmoon — APPROVED** [review]
Looks great!

**Comment by jsnmoon** [comment]
image

I've attached a screenshot to illustrate something we could fix in a follow-up PR. It doesn't make sense to shade cells with no data.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7687064).

**Pull request merged**